src/pages/accomodation/components/Reviews.tsx

```diff
@@ -0,0 +1,118 @@
+import Grid from '@mui/material/Grid';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogContent from '@mui/material/DialogContent';
+import IconButton from '@mui/material/IconButton';
+import CloseIcon from '@mui/icons-material/Close';
+import KeyboardArrowLeftRoundedIcon from '@mui/icons-material/KeyboardArrowLeftRounded';
+import Typography from '@mui/material/Typography';
+import Stack from '@mui/material/Stack';
+import Rating from '@mui/material/Rating';
+import Box from '@mui/material/Box';
+
+import useTheme from '@mui/material/styles/useTheme';
+import useMediaQuery from '@mui/material/useMediaQuery';
+import { useCallback, useState } from 'react';
+
+import { Review } from './Review';
+import useGetReviewsQuery from '../../../api/queries/reviews/useGetReviewsQuery';
+import LoadingPrimary from '../../../components/loader/LoadingPrimary';
+
+export const Reviews = ({ accommodationId }: { accommodationId: string }) => {
+  const { data: reviews, isError, isLoading } = useGetReviewsQuery(accommodationId);
+  const [open, setOpen] = useState(false);
+
+  const theme = useTheme();
+  const mobileScreen = useMediaQuery(theme.breakpoints.down('md'));
+
+  const toggleModal = useCallback(() => {
+    setOpen((prev) => !prev);
+  }, []);
+
+  if (isLoading || isError) return <LoadingPrimary />;
+
+  return reviews?.data?.length ? (
+    <Box>
+      <Stack
+        alignItems={'center'}
+        pb={12}
+        borderBottom={'1px solid'}
+        borderColor={'secondary2.light'}
+        mb={12}
+      >
+        <Typography fontSize={'2rem'}>Rating</Typography>
+        <Typography fontSize={'3.5rem'} fontWeight={600}>
+          {reviews.averageRate.toFixed(1)}
+        </Typography>
+        <Rating
+          readOnly
+          value={reviews.averageRate}
+          precision={0.1}
+          sx={{
+            '& .MuiRating-iconFilled': {
+              color: 'secondary.main',
+            },
+            'fontSize': '2rem',
+            'flexShrink': 0,
+          }}
+        />
+      </Stack>
+
+      <Grid container spacing={{ xs: 8, lg: 16 }}>
+        {reviews.data.slice(0, 4).map((review) => (
+          <Grid key={review.id} item xs={12} lg={6}>
+            <Review review={review} />
+          </Grid>
+        ))}
+      </Grid>
+
+      <Button
+        onClick={toggleModal}
+        variant={'outlined'}
+        sx={{
+          display: 'block',
+          mt: { xs: 8, lg: 16 },
+          mr: 'auto',
+          fontWeight: 600,
+          textTransform: 'none',
+        }}
+      >
+        Show all {reviews?.totalCount} reviews
```

Comment:
Sometimes you put optional chaining, sometimes not

think about what values can be empty and put optional chaining wisely

Reply:
At the time I made pr, property "totalCount" was fully functional in backend so I made additional check. Now it works properly and I removed optional chaining from this line.